From 7bcb97382f2f347dc348c3c31416ae7fd760ce6b Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@redhat.com>
Date: Fri, 17 Mar 2023 15:39:52 -0500
Subject: [PATCH] REGRESSION(261320@main): [GLib] Broke
 WebKitUserContentManager::script-message-received
 https://bugs.webkit.org/show_bug.cgi?id=254089

Reviewed by NOBODY (OOPS!).

I forgot to update the marshaller used by
WebKitUserContentManager::script-message-received. It worked perfectly
fine in my development environment for whatever reason, but was broken
in at least Ephy Tech Preview.

* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
---
 Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp b/Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp
index af2d963bb3b3..f778b19bc816 100644
--- a/Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp
+++ b/Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp
@@ -106,7 +106,11 @@ static void webkit_user_content_manager_class_init(WebKitUserContentManagerClass
             G_TYPE_FROM_CLASS(gObjectClass),
             static_cast<GSignalFlags>(G_SIGNAL_RUN_LAST | G_SIGNAL_DETAILED),
             0, nullptr, nullptr,
+#if ENABLE(2022_GLIB_API)
+            g_cclosure_marshal_VOID__OBJECT,
+#else
             g_cclosure_marshal_VOID__BOXED,
+#endif
             G_TYPE_NONE, 1,
 #if ENABLE(2022_GLIB_API)
             JSC_TYPE_VALUE);
